src/sentry/tasks/unmerge.py

```diff
@@ -249,6 +249,10 @@ def truncate_denormalizations(group):
         group_id=group.id,
     ).delete()
 
+    GroupEnvironment.objects.filter(
+        group_id=group.id,
+    ).delete()
```

Comment:
I just realized that I need to remove these one by one and ensure they are deleted from the cache, otherwise these are going to cause issues with the first seen rule condition.

Reply:
There's also a race condition here that between truncation and this part of the task executing that some events might be marked as first seen that weren't actually first seen. I'm not sure how big of a problem that is. I'm going to spend a few minutes trying to think about how to avoid that before revisiting this.